wsmaster/codenvy-hosted-system-ram/src/main/java/com/codenvy/service/systemRam/DockerBasedSystemRamLimitChecker.java

```diff
@@ -0,0 +1,141 @@
+        Future<SystemInfo> future = executor.submit(task);
+        SystemInfo systemInfo;
+        try {
+            systemInfo = future.get(5, SECONDS);
+            String[][] systemValues = systemInfo.getDriverStatus() == null ? systemInfo.getSystemStatus() : systemInfo.getDriverStatus();
+            if (systemValues == null) {
+                LOG.error("Empty system information was received from docker. All info from docker: {}", systemInfo);
+                return limitStatus;
+            }
+            int i = 0;
+            for (int l = systemValues.length; i < l; i++) {
+                String[] systemInfoEntry = systemValues[i];
+                if (systemInfoEntry.length == 2 && " └ Reserved Memory".equals(systemInfoEntry[0])) {
+                    allNodesRamUsage.add(systemInfoEntry[1]);
+                }
+            }
+        } catch (TimeoutException e) {
+            LOG.error("Timeout was reached while requesting to docker", e);
+            return limitStatus;
+        } catch (InterruptedException | ExecutionException e) {
+            LOG.error("A problem occurred while getting system information from docker", e);
+            return limitStatus;
+        }
+        for (String nodeRamUsage : allNodesRamUsage) {
+            String[] ramValues = nodeRamUsage.split(" / ");
+            if (ramValues.length != 2) {
+                LOG.error("A problem occurred while parsing system information from docker. " +
+                          "Expected: <used ram size> / <total ram size> but got: " + nodeRamUsage);
+                return limitStatus;
+            }
+            systemRamUsed += Size.parseSize(ramValues[0]);
+            systemRamTotal += Size.parseSize(ramValues[1]);
+        }
+        if (systemRamTotal * 0.9 < systemRamUsed) {
+            limitStatus = SYSTEM_RAM_LIMIT_EXCEEDED;
+            if (!systemRamLimitExceedMessageSent) {
+                sendRamLimitStatusMessage(SYSTEM_RAM_LIMIT_EXCEEDED.toString());
+                systemRamLimitExceedMessageSent = true;
+            }
+        } else {
+            limitStatus = SYSTEM_RAM_LIMIT_NOT_EXCEEDED;
+        }
+        return limitStatus;
+    }
+
+    private void sendRamLimitStatusMessage(String status) {
+        final ChannelBroadcastMessage bm = new ChannelBroadcastMessage();
+        bm.setChannel("system_ram_chanel");
+        bm.setBody(JsonUtils.getJsonString(status));
+        try {
+            WSConnectionContext.sendMessage(bm);
+        } catch (EncodeException | IOException e) {
+            LOG.error("A problem occurred while sending websocket message", e);
+        }
+    }
+
+    @ScheduleDelay(initialDelay = 1, delay = 1, unit = MINUTES)
+    private void freeRamCheckScheduler() {
+        if (systemRamLimitExceedMessageSent && SYSTEM_RAM_LIMIT_NOT_EXCEEDED != getSystemRamLimitStatus()) {
+            sendRamLimitStatusMessage(SYSTEM_RAM_LIMIT_NOT_EXCEEDED.toString());
```

Comment:
for me it is not obvious why we should send message about `SYSTEM_RAM_LIMIT_NOT_EXCEEDED` when `SYSTEM_RAM_LIMIT_NOT_EXCEEDED != getSystemRamLimitStatus()` says otherwise